modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/A11y.tsx

```diff
@@ -12,185 +12,214 @@
  * details.
  */
 
-import './A11y.scss';
-
-import ClayIcon from '@clayui/icon';
-import ClayList from '@clayui/list';
-import ClayPopover from '@clayui/popover';
-import {ReactPortal} from '@liferay/frontend-js-react-web';
-import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
-
-import useA11y from './useA11y';
+import React, {useCallback, useLayoutEffect, useState} from 'react';
+
+import {ErrorBoundary} from './components/ErrorBoundary';
+import {NotFound} from './components/NotFound';
+import Occurrence from './components/Occurrence';
+import {StackNavigator} from './components/StackNavigator';
+import Violation from './components/Violation';
+import {ViolationPopover} from './components/ViolationPopover';
+import Violations from './components/Violations';
+import useA11y from './hooks/useA11y';
+import {TYPES, useFilterViolations} from './hooks/useFilterViolations';
+import useIframeA11yChannel from './hooks/useIframeA11yChannel';
+import {Kind} from './hooks/useIframeClient';
 
 import type {A11yCheckerOptions} from './A11yChecker';
-import type {Violation as TViolation} from './useA11y';
-
-const rectAttrs: Array<keyof DOMRect> = [
-	'bottom',
-	'height',
-	'left',
-	'right',
-	'top',
-	'width',
-];
-
-interface IRectState {
-	rect: DOMRect | undefined;
-	hasRectChanged: boolean;
-}
 
-const DOMRectStub = {} as DOMRect;
+type Params = {
+	name?: string;
+	ruleId?: string;
+	target?: string;
+};
 
-const rectChanged = (a: DOMRect, b: DOMRect = DOMRectStub) =>
-	rectAttrs.some((prop) => a[prop] !== b[prop]);
+type IframePayload = {
+	ruleId: string;
+	target: string;
+};
 
-const useObserveRect = (
-	callback: (rect: DOMRect | undefined) => void,
-	node: Element | null
-) => {
-	const rafIdRef = useRef<number>();
+export function A11y(props: Omit<A11yCheckerOptions, 'callback'>) {
+	const violations = useA11y(props);
 
-	const run = useCallback(
-		(node: Element, state: IRectState) => {
-			const newRect = node.getBoundingClientRect();
+	const [state, dispatch] = useFilterViolations(violations);
 
-			if (rectChanged(newRect, state.rect)) {
-				state.rect = newRect;
+	const [params, setParams] = useState<Params>();
 
-				callback(state.rect);
-			}
+	const [activePage, setActivePage] = useState(0);
 
-			rafIdRef.current = window.requestAnimationFrame(() =>
-				run(node, state)
-			);
-		},
-		[callback]
-	);
+	const nodes = Object.keys(violations.nodes);
 
-	useEffect(() => {
-		if (node) {
-			run(node, {
-				hasRectChanged: false,
-				rect: undefined,
-			});
+	useLayoutEffect(() => {
+		if (nodes.length > 0) {
+			document.body.classList.add('a11y-body');
```

Comment:
I saw a11y-body class definition in Violation Sass file but I didn't understood what it does. Could you bring more details? maybe provide a better naming for it

Reply:
well I don't know I would have a better name for it, because this is a support-only class for adding `padding` to the body, so we can add the Panel on the right and the content will fit.

```scss
$a11y-panel-width: 320px;

.a11y-body {
	padding-right: $a11y-panel-width;

	.multi-panel-sidebar,
	.page-editor__sidebar {
		right: $a11y-panel-width;
	}

       // ....
}
```

For example, the first line will just add the `320px` space to the right of the body, and the other support classes we need to adjust manually because these components are rendered with `position: fixed;` or `position: absolute;` and there will be no effect on the `padding` on the body, I commented on this in the detail of the PR that integrates this. https://github.com/liferay-frontend/liferay-portal/pull/1188#issue-676722620